src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,39 @@
+package seedu.addressbook.data.person;
+
+/**
+ * Represents a Person's contact detail in the address book.
+ * Guarantees: immutable
+ */
+public class Contact {
+
+    public final String value;
+    protected boolean isPrivate;
+    
+    public Contact(String contact, boolean isPrivate) {
+        this.isPrivate = isPrivate;
+        String trimmedContact = contact.trim();
+        this.value = trimmedContact;
```

Comment:
Glad to see sanitising of the input.
Perhaps you could improve code quality by `this.value` = contact.trim()` instead of creating a temporary holding variable?